### DISCUSSION TITLE
lexer type high lighting in c

### DISCUSSION THREAD

**Discussion by YahYahDev** [orbitalquark/textadept#532] [Q&A] [answered]
the default configuration for the ansi_c lexer seems to not highlight user made types does any one know a way i could achieve that behavior? only way i can think of is some how appending the word list for types and adding user made types to it any ideas/input would be much appreciated!

**Comment by orbitalquark** [comment]
The lexer does not keep track of user-defined types, unfortunately. While you could have user-defined keyword lists in Textadept, there is no documented way to do so, and you'd typically only set the list once when the lexer is loaded, and not update it as the user edits the file.

Let me know if you'd like some documentation for this and I'll put it on my TODO list.

**Reply by YahYahDev** [reply]
yeah that would be awesome! ill switch over to nightly later this weekend probably so i can get the most up to date changes ect.

**Reply by orbitalquark** [reply, marked as answer]
You can put something like the following in your *~/.textadept/init.lua* to add to the list of types in the C lexer:

```
events.connect(events.LEXER_LOADED, function(name)
	if name == 'ansi_c' then buffer.lexer:set_word_list(lexer.TYPE, {'foo', 'bar', 'baz'}, true) end
end)
```

`buffer.lexer` is not documented, but `lexer:set_word_list()` is (https://orbitalquark.github.io/textadept/api.html#lexer.set_word_list).

**Reply by YahYahDev** [reply]
i could make a work around with that probably, ill give an update when i got somthing working from that idea!

**Reply by YahYahDev** [reply]
ran into an issue of the lua interpreter that text adept uses not allowing me to use variables in scope properly please see https://github.com/orbitalquark/textadept/issues/534

**Comment by YahYahDev** [comment]
made a module to highlight typedefs in a project with ctags
https://github.com/YahYahDev/TA-C-Type-Lexer
